x-pack/plugins/apm/server/lib/correlations/process_significant_term_aggs.ts

```diff
@@ -12,43 +12,65 @@ import {
 } from '../../../../../typings/elasticsearch/aggregations';
 
 export interface TopSigTerm {
-  bgCount: number;
-  fgCount: number;
   fieldName: string;
   fieldValue: string | number;
   score: number;
+  impact: number;
+  fieldCount: number;
+  valueCount: number;
 }
 
 type SigTermAgg = AggregationResultOf<
   { significant_terms: AggregationOptionsByType['significant_terms'] },
   {}
 >;
 
+function getMaxImpactScore(scores: number[]) {
+  if (scores.length === 0) {
+    return 0;
+  }
+
+  const sortedScores = scores.sort((a, b) => b - a);
+  const maxScore = sortedScores[0];
+
+  // calculate median
+  const halfSize = scores.length / 2;
+  const medianIndex = Math.floor(halfSize);
+  const medianScore =
+    medianIndex < halfSize
+      ? sortedScores[medianIndex]
+      : (sortedScores[medianIndex - 1] + sortedScores[medianIndex]) / 2;
+
+  return Math.max(maxScore, medianScore * 2);
+}
+
 export function processSignificantTermAggs({
   sigTermAggs,
-  thresholdPercentage,
 }: {
   sigTermAggs: Record<string, SigTermAgg>;
-  thresholdPercentage: number;
 }) {
   const significantTerms = Object.entries(sigTermAggs).flatMap(
     ([fieldName, agg]) => {
       return agg.buckets.map((bucket) => ({
         fieldName,
         fieldValue: bucket.key,
-        bgCount: bucket.bg_count,
-        fgCount: bucket.doc_count,
+        fieldCount: agg.doc_count,
+        valueCount: bucket.doc_count,
```

Comment:
Not sure about this change. Can you elaborate a bit?

Reply:
the previous `bgCount` wasn't needed since it represents only values for that field/value pair, when we really wanted to express the percentage in terms of the total number of background documents the aggregation applies to. E.g. `user_agent.name: HeadlessChrome` accounts for only 8% of all documents that meet the 75% threshold.

Reply:
previously it showed something like "80% of documents with `user_agent.name: HeadlessChrome` exist in the 75th percentile threshold", so it's a question of what is easier to understand for the user.